usecases/objects/references_delete.go

```diff
@@ -59,12 +58,34 @@ func (m *Manager) DeleteObjectReference(ctx context.Context, principal *models.P
 		}
 	}
 
+	res, err := m.getObjectFromRepo(ctx, input.Class, input.ID, additional.Properties{}, nil, tenant)
+	if err != nil {
+		errnf := ErrNotFound{}
+		if errors.As(err, &errnf) {
+			return &Error{"source object", StatusNotFound, err}
+		} else if errors.As(err, &ErrMultiTenancy{}) {
+			return &Error{"source object", StatusUnprocessableEntity, err}
+		}
+
+		return &Error{"source object", StatusInternalServerError, err}
+	}
+	input.Class = res.ClassName
+
+	if err := validateReferenceName(input.Class, input.Property); err != nil {
+		return &Error{err.Error(), StatusBadRequest, err}
+	}
+
+	class, schemaVersion, _, typedErr := m.getAuthorizedFromClass(ctx, principal, input.Class)
+	if typedErr != nil {
+		return typedErr
+	}
```

Comment:
Shouldn't the `getAuthorizedFromClass` call be made before the `getObjectFromRepo` call? Otherwise users could mine information about which objects exist or not. I recall this issue in terms of compliance, maybe it isn't a problem anymore? wdyt?